### PR TITLE
Na ps duplicate error

### LINKED ISSUE #9: 9. As a user, I want to be alerted when I’m entering an empty item or an item that's already on my list.
## Summary
Users shouldn't be able to add an empty item to their list, or add the same item twice. If they try to do this, we need to show them an error message that explains the problem. This way, we’ll prevent some clutter in their lists.

## Acceptance criteria
- [x] Show an error message if the user tries to submit an empty item
- [x] Show an error message if the user tries to submit a new item that is _identical_ to an existing name. For instance, if the list contains `apples` and the user adds `apples`.
- [x] Show an error message if the user tries to submit a new item that matches an existing name with punctuation and casing normalized. For instance, if the list contains `apples` and the user adds `aPples` or `apples,` or `a pples`.
- [x] The user’s original input is saved in the database

##  Notes
- There are a couple of ways to approach this task. If you’re not sure how to begin, _ask your mentors_ for advice!

### PR DESCRIPTION
**Pull request by snowsneakers**
## Description

This code will display an error if the user enters an item already stored in the data base. This even work if there are spaces, special characters, or capitals. We did this by first setting the user input to lowercase by using the `.toLowerCase()` method. We then `.split("")` to turn the string into an array. Next, we filtered the user input by `.filter(x => !specialChar.includes(x) && x !== ' '`. This will filter out any special characters in the user input. The `specailchar` variable is an array of all specail characters. If there is an item in the data base that equals the user input an error is displayed. If there isn't, the user input would be added to the list. We scratched our head for awhile trying to find a more simple, cleaner regex alternative, but we decided to just go with this.

We also moved the add item form reset function out of the setTimeout to clear the input immediately after the user submits instead of 2 seconds later.

## Related Issue

Closes #9 

## Acceptance Criteria

- [x] Show an error message if the user tries to submit an empty item
- [x] Show an error message if the user tries to submit a new item that is _identical_ to an existing name. For instance, if the list contains `apples` and the user adds `apples`.
- [x] Show an error message if the user tries to submit a new item that matches an existing name with punctuation and casing normalized. For instance, if the list contains `apples` and the user adds `aPples` or `apples,` or `a pples`.
- [x] The user’s original input is saved in the database

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

### Before

issue-9-before

### After

issue-9-after

## Testing Steps / QA Criteria

1. Add an item like "milk"
2. Try to add another item containing spaces, caps, and special characters like "M il K!"
3. Any variation of spaces, caps and special characters should show "item already in the list"

### CODE REVIEW

File: src/views/AddItem.jsx [hunk 1]
```diff
@@ -1,31 +1,76 @@
...
+				.split('')
+				.filter((x) => !specialChar.includes(x) && x !== ' ')
+				.join('') ===
+			itemName
+				.toLowerCase()
+				.split('')
+				.filter((x) => !specialChar.includes(x) && x !== ' ')
+				.join(''),
+	);
+
```
Comment (snowsneakers) [inline src/views/AddItem.jsx:52]: There must be another way to do this

File: src/views/AddItem.jsx [hunk 2]
```diff
@@ -1,31 +1,76 @@
...
+				itemName: '',
+			}));
 			setTimeout(() => {
-				setFormData((prevState) => ({
-					...prevState,
-					itemName: '',
-				}));
 				setMessage('');
+				setDuplicateError(false);
 			}, 2000);
```
Comment (meera-ramesh19) [inline src/views/AddItem.jsx:74]: Great idea to move the add item form reset function out of the setTimeout

**Review by meera-ramesh19 — APPROVED** [review]
Looks great

File: src/views/AddItem.jsx (hunk 1 above)
Comment (kaitlinLewis) [reply src/views/AddItem.jsx:52]: I am so surprised that there is not a special character method already! On a cellphone, there are so many more special characters eg money, math, copyright etc as well as international users would have different keyboards. I agree with you Pat. It does feel like there should be a more encompassing way.

**Review by kaitlinLewis — APPROVED** [review]

**Comment by daredtech** [comment]
Hi team, for `Show an error message if the user tries to submit an empty item`: 
if the message is empty it works, however, if I enter a space and submit, the item goes through:

Screen Shot 2022-11-04 at 8 59 31 AM

Perhaps it is something you may want to address as well.

**Comment by NuriyaAkh** [comment]
> Hi team, for `Show an error message if the user tries to submit an empty item`: if the message is empty it works, however, if I enter a space and submit, the item goes through:
> 
> Screen Shot 2022-11-04 at 8 59 31 AM
> 
> Perhaps it is something you may want to address as well.

Thank you for catching this sneaky bug, Pat and I working on it

**Review by cafloyd — APPROVED** [review]
Fantastic work, you two! Looks like all the feedback on this PR has been addressed, I think it's ready to merge!

**Pull request merged**